### PR TITLE
Move Network to src

### PR DESCRIPTION
**Pull request by zeroadam**
Move network class to src

Related to #3878

### CODE REVIEW

**Comment by zeroadam** [comment]
Blocks https://github.com/friendica/friendica-addons/pull/504

File: src/Util/Network.php [hunk 1]
```diff
@@ -0,0 +1,1087 @@
...
+		return false;
+	}
+
+	/**
+	 * @brief Checks that email is an actual resolvable internet address
+	 *
+	 * @param string $addr The email address
+	 * @return boolean True if it's a valid email address, false if it's not
+	 */
+	public static function validateEmail($addr)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::isEmailDomainValid()`

File: src/Util/Network.php [hunk 2]
```diff
@@ -0,0 +1,1087 @@
...
+	/**
+	 * @brief Check if URL is allowed
+	 *
+	 * Check $url against our list of allowed sites,
+	 * wildcards allowed. If allowed_sites is unset return true;
+	 *
+	 * @param string $url URL which get tested
+	 * @return boolean True if url is allowed otherwise return false
+	 */
+	public static function allowedURL($url)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::isUrlAllowed()`

File: src/Util/Network.php [hunk 3]
```diff
@@ -0,0 +1,1087 @@
...
+
+	/**
+	 * Checks if the provided url domain is on the domain blocklist.
+	 * Returns true if it is or malformed URL, false if not.
+	 *
+	 * @param string $url The url to check the domain from
+	 *
+	 * @return boolean
+	 */
+	public static function blockedURL($url)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::isUrlBlocked()`

File: src/Util/Network.php [hunk 4]
```diff
@@ -0,0 +1,1087 @@
...
+	/**
+	 * @brief Check if email address is allowed to register here.
+	 *
+	 * Compare against our list (wildcards allowed).
+	 *
+	 * @param  string $email email address
+	 * @return boolean False if not allowed, true if allowed
+	 *    or if allowed list is not configured
+	 */
+	public static function allowedEmail($email)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::isEmailDomainAllowed()`

File: src/Util/Network.php [hunk 5]
```diff
@@ -0,0 +1,1087 @@
...
+
+	/**
+	 * Checks for the existence of a domain in a domain list
+	 *
+	 * @brief Checks for the existence of a domain in a domain list
+	 * @param string $domain
+	 * @param array  $domain_list
+	 * @return boolean
+	 */
+	public static function allowedDomain($domain, array $domain_list)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::isDomainAllowed()`

File: src/Util/Network.php [hunk 6]
```diff
@@ -0,0 +1,1087 @@
...
+			if (fnmatch($pat, $domain) || ($pat == $domain)) {
+				$found = true;
+				break;
+			}
+		}
+
+		return $found;
+	}
+
+	public static function avatarImg($email)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::lookupAvatarByEmail()`

File: src/Util/Network.php [hunk 7]
```diff
@@ -0,0 +1,1087 @@
...
+
+		if (! $avatar['success']) {
+			$avatar['url'] = System::baseUrl() . '/images/person-175.jpg';
+		}
+
+		logger('Avatar: ' . $avatar['email'] . ' ' . $avatar['url'], LOGGER_DEBUG);
+		return $avatar['url'];
+	}
+
+	public static function parseXmlString($s, $strict = true)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `Util\XML::parseString()`.

File: src/Util/Network.php [hunk 8]
```diff
@@ -0,0 +1,1087 @@
...
+			logger('libxml: parse: error: ' . $s, LOGGER_DATA);
+			foreach (libxml_get_errors() as $err) {
+				logger('libxml: parse: ' . $err->code." at ".$err->line.":".$err->column." : ".$err->message, LOGGER_DATA);
+			}
+			libxml_clear_errors();
+		}
+		return $x;
+	}
+
+	public static function scaleExternalImages($srctext, $include_link = true, $scale_replace = false)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `Content\Text\BBCode::scaleExternalImages()`

File: src/Util/Network.php [hunk 9]
```diff
@@ -0,0 +1,1087 @@
...
+				}
+			}
+		}
+
+		// replace the special char encoding
+		$s = htmlspecialchars($s, ENT_NOQUOTES, 'UTF-8');
+		return $s;
+	}
+
+	public static function fixContactSslPolicy(&$contact, $new_policy)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `Model\Contact::updateSslpolicy()` and change the by-reference parameter to a simple return.

File: src/Util/Network.php [hunk 10]
```diff
@@ -0,0 +1,1087 @@
...
+	 * @todo Remove the $fetchbody parameter that generates an extraneous HEAD request
+	 *
+	 * @see ParseUrl::getSiteinfo
+	 *
+	 * @param string $url       A user-submitted URL
+	 * @param int    $depth     The current redirection recursion level (internal)
+	 * @param bool   $fetchbody Wether to fetch the body or not after the HEAD requests
+	 * @return string A canonical URL
+	 */
+	public static function originalURL($url, $depth = 1, $fetchbody = false)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::finalUrl()`

File: src/Util/Network.php [hunk 11]
```diff
@@ -0,0 +1,1087 @@
...
+						return(Network::originalURL(substr($value, 4), ++$depth));
+					}
+				}
+			}
+		}
+
+		return $url;
+	}
+
+	public static function shortLink($url)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::shortenUrl()`

File: src/Util/Network.php [hunk 12]
```diff
@@ -0,0 +1,1087 @@
...
+	/**
+	 * @brief Encodes content to json
+	 *
+	 * This function encodes an array to json format
+	 * and adds an application/json HTTP header to the output.
+	 * After finishing the process is getting killed.
+	 *
+	 * @param array $x The input content
+	 */
+	public static function jsonReturnAndDie($x)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::jsonExit()`

File: src/Util/Network.php [hunk 13]
```diff
@@ -0,0 +1,1087 @@
...
+	}
+
+	/**
+	 * @brief Find the matching part between two url
+	 *
+	 * @param string $url1
+	 * @param string $url2
+	 * @return string The matching part
+	 */
+	public static function matchingURL($url1, $url2)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::getUrlMatch()`

File: src/Util/Network.php [hunk 14]
```diff
@@ -0,0 +1,1087 @@
...
+	}
+
+	/**
+	 * @brief Glue url parts together
+	 *
+	 * @param array $parsed URL parts
+	 *
+	 * @return string The glued URL
+	 */
+	public static function unParseURL($parsed)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please rename to `Network::unparseUrl()`

**Comment by zeroadam** [comment]
Hahaha! Will do :smile:

File: src/Util/Network.php [hunk 15]
```diff
@@ -0,0 +1,967 @@
...
+
+		return $body;
+	}
+
+	/**
+	 * Generic XML return
+	 * Outputs a basic dfrn XML status structure to STDOUT, with a <status> variable
+	 * of $st and an optional text <message> of $message and terminates the current process.
+	 */
+	public static function xmlExit($st, $message = '')
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `System::xmlExit()`.

File: src/Util/Network.php [hunk 16]
```diff
@@ -0,0 +1,967 @@
...
+
+	/**
+	 * @brief Send HTTP status header and exit.
+	 *
+	 * @param integer $val         HTTP status result value
+	 * @param array   $description optional message
+	 *                             'title' => header title
+	 *                             'description' => optional message
+	 */
+	public static function httpStatusExit($val, $description = [])
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `System::httpExit()`.

File: src/Util/Network.php [hunk 17]
```diff
@@ -0,0 +1,967 @@
...
+	/**
+	 * @brief Encodes content to json
+	 *
+	 * This function encodes an array to json format
+	 * and adds an application/json HTTP header to the output.
+	 * After finishing the process is getting killed.
+	 *
+	 * @param array $x The input content
+	 */
+	public static function jsonExit($x)
```
Comment (MrPetovan) [inline src/Util/Network.php]: Please move to `System::jsonExit()`.

**Pull request merged**